plugin/pkg/shoot/validator/admission.go

```diff
@@ -534,6 +538,14 @@ func (c *validationContext) validateProvider() field.ErrorList {
 		return allErrs
 	}
 
+	// TODO (voelzmo): remove this 'if' statement once we gave owners of existing Shoots a nice grace period to move away from 'internal' apiVersion
+	if a.GetOperation() == admission.Create {
+		_, gvk, _ := serializer.NewCodecFactory(kubernetesscheme.Scheme).UniversalDecoder(corev1.SchemeGroupVersion).Decode(c.shoot.Spec.Provider.InfrastructureConfig.Raw, nil, nil)
```

Comment:
If done here, this check should be done not only for `Spec.Provider.InfrastructureConfig`, but for all fields that are of type `RawExtension` and may contain provider configuration. These are (but please double-check if I haven't missed anything):
* `Spec.Provider.ControlPlaneConfig`
* `Spec.Extensions[].ProviderConfig`
* `Spec.Networking.ProviderConfig`
* `Spec.Provider.Workers[].ProviderConfig`
* `Spec.Provider.Workers[].Machine.Image.ProviderConfig`
* `Spec.Provider.Workers[].CRI.ContainerRuntimes[].ProviderConfig`

Reply:
Thanks for compiling this list! I was thinking about this while writing up the PR summary – therefore the comment above in "Special notes for your reviewer". Given that this piece of code is here, adding additional checks for these should be possible with not too much additional efforts. Do you think we should do this as part of this PR, or rather focus here on the InfrastructureConfig bit?

Reply:
I think it should all be done as part of this PR, there is no need to focus on `InfrastructureConfig`. But let's first clarify the comment by @rfranzke about whether this should be done here or in the extensions themselves.

Reply:
done